lib/mayaHydra/hydraExtensions/CMakeLists.txt

```diff
@@ -49,11 +49,14 @@ mayaHydra_compile_config(${TARGET_NAME})
 # include directories
 # -----------------------------------------------------------------------------
 target_include_directories(${TARGET_NAME} 
-            PUBLIC
+            SYSTEM PUBLIC
+                ${PXR_INCLUDE_DIRS}
                 ${MAYA_INCLUDE_DIRS}
+)
+target_include_directories(${TARGET_NAME}
+            PUBLIC
                 ${PYTHON_INCLUDE_DIR}
-                ${PXR_INCLUDE_DIRS}
-                ${BOOST_INCLUDE_DIR}
+                ${BOOST_INCLUDE_DIR}               
                 ${CMAKE_BINARY_DIR}/include
```

Comment:
Can't we fuse this into a single declaration?  Also, why are we using `SYSTEM PUBLIC`?  Is it required?

Reply:
Would be nice to comment on this in the CMakeLists.txt file.  And can we merge this into a single declaration to avoid duplication?

Reply:
I don't think so. From the cmake doc it appears that SYSTEM applies to everything you list after.

Reply:
@ppt-adsk did you mean something like this? :
```
target_include_directories(${TARGET_NAME} 
            SYSTEM PUBLIC
                ${PXR_INCLUDE_DIRS}
                ${MAYA_INCLUDE_DIRS}

            PUBLIC
                ${PYTHON_INCLUDE_DIR}
                ${BOOST_INCLUDE_DIR}               
                ${CMAKE_BINARY_DIR}/include
            PRIVATE
                $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
                $<$<BOOL:${MayaUsd_FOUND}>:${MAYAUSD_INCLUDE_DIR}>
)
```

Reply:
Yes, @seando-adsk doubts this will work, but I feel this is more maintainable if it does work.

Reply:
Maybe the PUBLIC section needs to go before the SYSTEM PUBLIC section.

Reply:
I don't think it will work.
https://cmake.org/cmake/help/v3.14/command/target_include_directories.html

Plus how will you test it? What you have above Rakesh will just put everything in SYSTEM.